src/descriptor/sortedmulti.rs

```diff
@@ -32,15 +32,21 @@ pub struct SortedMultiVec<Pk: MiniscriptKey, Ctx: ScriptContext> {
 }
 
 impl<Pk: MiniscriptKey, Ctx: ScriptContext> SortedMultiVec<Pk, Ctx> {
-    fn constructor_check(&self) -> Result<(), Error> {
+    fn constructor_check(mut self) -> Result<Self, Error> {
         // Check the limits before creating a new SortedMultiVec
         // For example, under p2sh context the scriptlen can only be
         // upto 520 bytes.
-        let term: Terminal<Pk, Ctx> = Terminal::Multi(self.inner.k(), self.inner.data().to_owned());
+        let term: Terminal<Pk, Ctx> = Terminal::Multi(self.inner);
         let ms = Miniscript::from_ast(term)?;
         // This would check all the consensus rules for p2sh/p2wsh and
         // even tapscript in future
-        Ctx::check_local_validity(&ms).map_err(From::from)
+        Ctx::check_local_validity(&ms).map_err(Error::from)?;
```

Comment:
```suggestion
        Ctx::check_local_validity(&ms)?;
```

Reply:
Yeah, I could drop it here. In future though I want to move away from using `from` and explicitly write variants (e.g. `map_err(Error::ScriptContext)` or whatever).

But agreed that `.map_err(Error::from)?` is the worst of both worlds and I should drop it.